### PR TITLE
Fixed #11 Implement ReplicationFilterCompiler for JavaScript

### LINKED ISSUE #11: Implement ReplicationFilterCompiler and ReplicationFilter
Would be great if you could also implement ReplicationFilterCompiler.

### PR DESCRIPTION
**Pull request by hideki**

### CODE REVIEW

File: src/main/java/com/couchbase/lite/javascript/JavaScriptReplicationFilterCompiler.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import com.couchbase.lite.ReplicationFilter;
+import com.couchbase.lite.ReplicationFilterCompiler;
+
+/**
+ * Created by hideki on 10/28/15.
+ */
+public class JavaScriptReplicationFilterCompiler implements ReplicationFilterCompiler {
+    @Override
+    public ReplicationFilter compileFilterFunction(String source, String language) {
+        return new ReplicationFilterBlockRhino(source);
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/javascript/JavaScriptReplicationFilterCompiler.java]: Do we need to check what the language is and throw exception if it's not javascript?

File: src/main/java/com/couchbase/lite/javascript/ReplicationFilterBlockRhino.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    private static WrapFactory wrapFactory = new CustomWrapFactory();
+    private Scriptable scope;
+    private GlobalScope globalScope;
+    private Function filterFunction;
+
+    public ReplicationFilterBlockRhino(String src){
+        org.mozilla.javascript.Context ctx = org.mozilla.javascript.Context.enter();
+        try {
+            ctx.setOptimizationLevel(-1);
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/javascript/ReplicationFilterBlockRhino.java]: What is the -1 meaning?

**Comment by pasin** [comment]
I think copyright header in the new files is missing. Could you please add that as well?

File: src/main/java/com/couchbase/lite/javascript/ReplicationFilterBlockRhino.java (hunk 2 above)
Comment (hideki) [inline src/main/java/com/couchbase/lite/javascript/ReplicationFilterBlockRhino.java]: https://developer.mozilla.org/en-US/docs/Mozilla/Projects/Rhino/Optimization
```
-1

Interpretive mode is always used. The compilation time is minimized at the expense of runtime performance. No class files are generated, which may improve memory usage depending on your system. Another benefit of the interpreted mode is that the interpreter performs tail-call elimination of recursive functions. Also, you must use this optimization level if your code uses Continuation objects.

If the optimization package is not available, then optimization acts as if it is always -1.
```

**Pull request merged**